### ISSUE TITLE
InMemoryBlobCache used by default using latest release package iOS Xamarin

### ISSUE THREAD

**Issue by Letibee** [akavache/Akavache#238]
I'm using Xamarin Studio and the latest version of akavache built for Unified iOS (V 4.1.0) However stored data does not persist app restarts.

<b>100% repro sample project</b> : https://dl.dropboxusercontent.com/u/21635985/InMemoryBlobCache.zip (You will need to restore packages)
Using : Latest release (4.1.0) taken from https://github.com/akavache/Akavache/releases

It's a very simple project. The iOS project is built using the downloaded Xamarin.iOS10 binaries. (Note: No SQL DLL's appear in the iOS folder download from https://github.com/akavache/Akavache/releases I'm not sure if that's relevant or not!)
The PCL is built using the portable binaries Portable-Net45+Win8+WP8+Wpa81 also obtained from https://github.com/akavache/Akavache/releases

I noticed no values are being retrieved after device reset. You can reproduce this easily using the attached project. After examining further, I noticed that everything seems to use the InMemoryBlobCache, I'm sure this is incorrect.

![screen shot 2015-08-12 at 21 43 26](https://cloud.githubusercontent.com/assets/12245291/9224149/3666c698-413b-11e5-8d30-e93b0c627479.png)

I guess the two issues are related, but I'm creating this issue here in the hopes that one of the developers can help me resolve this issue.

Again, I must stress that I'm using the official Akavache release here. and this is the second issue I've ran into out of the box. Maybe I'm doing something insanely stupid....

**Comment by TheTekton** [comment]
I can confirm the InMemoryBlobCache being used for all providers, with iOS.

**Comment by Letibee** [comment]
I guess no persistence would be understandable if I was doing something unique or different, but since I'm using the stock installation in the simplest of projects imaginable, I feel like something else is going wrong. (Or, I'm doing something incredibly silly, though I can't imagine that's the case, since I'm not trying to do anything other than use Akavache for it's most basic functionality.)

The fact that I've added a incredibly basic 100% report project to the issue, should help the devs locate the problem fairly easily. So, let's hope they're listening.

**Comment by flagbug** [comment]
@Letibee You have to reference Akavache.Sqlite3 from your iOS project, not just from the PCL project

**Comment by Letibee** [comment]
@flagbug That may or may not be the case, the iOS project does not have a direct reference to Akavache.Sqlite3, since there is no provided binary in Xamarin.iOS10

Note : This is the downloaded package from the release page : https://github.com/akavache/Akavache/releases

Regardless if there is a direct or indirect reference, the problem of data loss on app restart still exists with the current released binaries as is evident in the attached project. (Post 1)

**Comment by Letibee** [comment]
I've provided a 100% repro project that anyone can download and use. Download, restore and build, result == broken.

No ideas?

Akavache is working well, apart from this one bug. Data just doesn't persist between application runs.

**Comment by TheTekton** [comment]
I cloned Akavache locally, a month or so ago, to see if I could get the other providers to work with Unified iOS, in v4.1.0. I'm pretty sure it's an issue with SQLite, as flagbug was suggesting. However, I'm thinking, from recollection, that it's more to do with an issue with SQLitePCL.raw_basic. Haven't had time to tinker more yet, but I did download your zip earlier (maybe can tinker over the weekend). I'm pretty sure this issue would be fairly straight forward to fix, if I was more familiar with those bits, which I'm not really, yet. @paulcbetts If it's not too much trouble to ask, is there anything blatantly obvious we're missing here?
